src/Shared/CommunicationsUtilities.cs

```diff
@@ -655,7 +655,9 @@ internal static void Trace(int nodeId, string format, params object[] args)
         /// <returns>Base Handshake</returns>
         private static long GetBaseHandshakeForContext(TaskHostContext hostContext)
         {
-            string salt = Environment.GetEnvironmentVariable("MSBUILDNODEHANDSHAKESALT");
+            string salt = Environment.GetEnvironmentVariable("MSBUILDNODEHANDSHAKESALT") +
+                Environment.GetEnvironmentVariable("DOTNET_HOST_PATH") +
+                Environment.GetEnvironmentVariable("DOTNET_ROOT");
```

Comment:
Should we be using environment variables for these, or determining them programmatically? I expected the latter, but if the env var is sufficient that's easy . . .

Reply:
You mentioned they were environment variables in the issue. I can try to figure them out another way if you'd prefer. I just set them as desired on my computer.

Reply:
We shouldn't require user intervention to set these. I believe `dotnet build` and `dotnet msbuild` both set them for the child MSBuild process. Can you confirm that? If they both do, using the env var is probably enough.

@dsplaisted do you have an opinion on this?

Reply:
Isn't the path to MSBuild.dll also part of the handshake?  In that case I don't think we would additionally need to add the path to `dotnet`, as that should be less unique than the MSBuild path.